File: src/main/java/Ex12_Set.java
import java.util.HashSet;
import java.util.Set;

import static java.util.Collections.emptySet;
import static java.util.Collections.singleton;

public class Ex12_Set {
    public static void main(String[] args) {
        Set<String> oneFruit = singleton("Kiwi"); // Declare singleton set of fruit => Immutable

        Set<String> fruits = new HashSet<>(); // Declare mutable set of fruits
        fruits.add("Apple");
        fruits.add("Banana");
        fruits.add("Orange");

        Set<String> otherFruits = Set.of("Avocado", "Ananas", "Cranberry"); // Immutable set of fruits

        Set<Object> noFruits = emptySet(); // Declare empty set
    }
}

File: src/main/kotlin/Ex12_Set.kt
fun main() {
  /* Set are Immutable on Kotlin */
  /* MutableSet are mutable on Kotlin */

  val oneFruitMutable: MutableSet<String> = mutableSetOf("Kiwi") // Mutable singleton set of fruit
  val oneFruitImmutable: Set<String> = setOf("Kiwi") // Immutable singleton set of fruit

  val fruits: MutableSet<String>  = mutableSetOf("Apple", "Banana", "Orange") // Mutable set of fruits

  val otherFruits: Set<String> = setOf("Avocado", "Ananas", "Cranberry") // Immutable set of fruits

  val noFruitMutable: MutableSet<String> = mutableSetOf() // Mutable empty set
  val noFruitImmutable: Set<String> = setOf() // Immutable empty set
}